addons/map/functions/fnc_switchFlashlight.sqf

```diff
@@ -16,10 +16,18 @@
 
 #include "script_component.hpp"
 
-params ["_flashlight"];
+params ["_newFlashlight"];
+private _oldFlashlight = GVAR(flashlightInUse);
```

Comment:
Yes, I think B would be the best option, although C wouldn't cause much network traffic either as you'd use that from time to time only.